gtp2ogs.js

```diff
@@ -1725,197 +1780,340 @@ class Connection {
+                    }
+                    if (universalMaintimeTimecontrolString === "canadian" && t.main_time > universalMaintimeNumber) {
+                        universalMaintimeTimecontrolSentence = "Main Time ";
+                        universalMaintimeEndingSentence = ".";
+                        conn_log(universalMaintimeConnSentence + universalMaintimeToString + " in " + universalMaintimeTimecontrolString);
+                        return { reject : true, msg:  `${universalMaintimeMinimumMaximumSentence} ${universalMaintimeTimecontrolSentence} ${universalMaintimeForRankedUnrankedSentence} ${universalMaintimeToString} ${universalMaintimeIncreaseDecreaseSentence} ${universalMaintimeTimecontrolSentence} ${universalMaintimeEndingSentence}` };
+                    } 
+                }
             }
         }
 
-        if (argv.maxmaintimeunranked && !notification.ranked) {
-            if (["simple","none"].indexOf(t.time_control) >= 0) {
-                conn_log("Maximum main time not supported in time control: " + t.time_control);
-                return { reject: true, msg: "Maximum main time not supported in time control " + t.time_control + ", please choose a time control that supports the use of a minimum main time, such as byoyomi,fischer,canadian. " };
-            }
-            if (t.total_time   > argv.maxmaintimeunranked  || // absolute
-                t.initial_time > argv.maxmaintimeunranked  || // fischer
-                t.max_time     > argv.maxmaintimeunranked  || // fischer
-                t.main_time    > argv.maxmaintimeunranked) {  // others
-                    let botSecondsTime = timespanToDisplayString(argv.maxmaintimeunranked);
-                    conn_log(user.username + " wanted main time unranked above maxmaintimeunranked " + botSecondsTime);
-                    return { reject: true, msg: "Maximum main time for unranked games is " + botSecondsTime + ", please reduce main time. \n - If you use canadian byo-yomi, set the average time per stone, for example 5 minutes/25 stones needs to be set up as 180 seconds (5 minutes) divided by 25 stones , which equals 7.2 seconds per stone. " };
-            }
-        }
+        ////// end of *** UHMAEAT : Universal Highly Modulable And Expandable Argv Tree ***
 
         if (argv.minperiods && (t.periods < argv.minperiods)) {
             conn_log(user.username + " wanted too few periods: " + t.periods);
-            return { reject: true, msg: "Minimum number of periods is " + argv.minperiods + " , please increase the number of periods, for example " + argv.minperiods + " x 30 seconds " };
+            return { reject: true, msg: "Minimum number of periods is " + argv.minperiods + " , please increase the number of periods" };
```

Comment:
More spaces to pay attention to here `" ,` => `",`

Reply:
ok but there are so much already... :sob: